Add support for image digests in helm charts

aanm opened the pull request:
Read per commit

aanm committed:
contrib/release: do not require images to be download locally

The SHAs are available in the GH run, so we can use the GH API to
retrieve them automatically.

Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
.github: generate release files automatically

Having files with the right format will make it easier to download them
locally and use them as part of the release process.

Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
install: add digests into helm charts

This commit adds a simple way to add the image digests into the official
helm charts.

Signed-off-by: André Martins <andre@cilium.io>

joestringer reviewed (approved):
Same comments as the [v1.9](https://github.com/cilium/cilium/pull/15186) review, all non-blocking nits.

aanm committed:
.github: remove comments of GH action versions

Dependabot will update those commit SHAs automatically and the version
number will quickly outdated. To avoid confusion, we will remove the
comments related with the versions a SHA points to from all GH actions.

Signed-off-by: André Martins <andre@cilium.io>
